x-pack/plugins/fleet/server/routes/agent/handlers.ts

```diff
@@ -308,17 +310,32 @@ export const getAgentDataHandler: RequestHandler<
 > = async (context, request, response) => {
   const coreContext = await context.core;
   const esClient = coreContext.elasticsearch.client.asCurrentUser;
-
-  const returnDataPreview = request.query.previewData;
-  const agentIds = isStringArray(request.query.agentsIds)
+  const agentsIds = isStringArray(request.query.agentsIds)
     ? request.query.agentsIds
     : [request.query.agentsIds];
+  const { pkgName, pkgVersion, previewData: returnDataPreview } = request.query;
+
+  // If a package is specified, get data stream patterns for that package
+  // and scope incoming data query to that pattern
+  let dataStreamPattern: string | undefined;
+  if (pkgName && pkgVersion) {
+    const packageInfo = await getPackageInfo({
+      savedObjectsClient: coreContext.savedObjects.client,
+      prerelease: true,
+      pkgName,
+      pkgVersion,
+    });
+    dataStreamPattern = (packageInfo.data_streams || [])
```

Comment:
I need to remember this pattern next time I map a possibly undefined attribute.  :thumbs-up